tests/FakeItEasy.Tests/TestHelpers/ExpressionHelper.cs

```diff
@@ -26,6 +26,12 @@ public static IInterceptedFakeObjectCall CreateFakeCall<TFake>(Expression<Action
             return CreateFakeCall(A.Fake<TFake>(), callSpecification);
         }
 
+        public static MethodInfo GetMethodInfo<T>(Expression<Action<T>> callSpecification) =>
+            GetMethodInfo((LambdaExpression)callSpecification);
+
+        public static MethodInfo GetMethodInfo<T, TResult>(Expression<Func<T, TResult>> callSpecification) =>
+            GetMethodInfo((LambdaExpression)callSpecification);
+

```

Comment:
1. I tried `Expression<Func<T, object?>>`. It fell down when there was boxing.
    I just now extended the method to peel the conversions off. It's a minor change, actually, but maybe your other idea is better? On that…
2. I am not smart enough to do this, I think. If it pleases you, I would love an example.

Reply:
> I am not smart enough to do this, I think. If it pleases you, I would love an example.

Sorry, my explanation was not very clear. I mean that instead of this:
```csharp
GetMethodInfo<string, int>(x => x.Length)
```
You could do this:
```csharp
GetMethodInfo((string x) => x.Length)
```
This way you only need to specify `T`, not `TResult`

Reply:
Oh. Yes. I both never would've figured this out, and should've figured it out. Thanks. Opting for this.